sdk/helper/tokenutil/tokenutil.go

```diff
@@ -274,6 +296,23 @@ func (t *TokenParams) PopulateTokenAuth(auth *logical.Auth) {
 	auth.TokenType = t.TokenType
 	auth.TTL = t.TokenTTL
 	auth.NumUses = t.TokenNumUses
+
+	if t.TokenStrictlyBindIP {
+		addrs := []string{req.Connection.RemoteAddr}
+		values, present := req.Headers[textproto.CanonicalMIMEHeaderKey("X-Forwarded-For")]
+		if present && len(values) > 0 {
+			addrs = values
```

Comment:
Are you sure this has the effect you're intending it to have? All IPs which are part of the `X-Forwarded-For` header will be added to the token's allowed CIDR list. So if OpenBao is behind some L7 load-balancer, it's IP will be added to the token and thus everyone with that token can access the cluster.

By convention the first IP in the list of IPs of the `X-Forwarded-For` header is the one of the original client. 
_If_ you want to rely on the header to get the client's source IP, I think it would be good to only add the first value of the header to the token CIDR list.
See the relevant [MDN page ](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For#syntax) for more information about this.

Reply:
@JanMa Aha!

Initially I thought `X-Forwarded-For` would need to be manually added as a [`passthrough_request_header`](https://janma.github.io/openbao/api-docs/system/mounts/#tune-mount-configuration), but even then, we should probably respect [`x_forwarded_for_hop_skips`](https://janma.github.io/openbao/docs/configuration/listener/tcp/) in the listener config...


...but now that I read `http.WrapForwardedForHandler` more closely (see the set of `r.RemoteAddr` just before  the last `ServeHTTP`), I realize that I can just take `req.RemoteAddr` verbatim and not deal with this, so we can remove this entirely. Thanks!